### PR TITLE
Add skip_on_exit_code to SSHOperator

### PR DESCRIPTION
**Pull request by dolfinus**
Now SSHOperator supports option `skip_on_exit_code`, just like `BashOperator`, `PythonVirtualenvOperator` and others.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/ssh/operators/ssh.py [hunk 1]
```diff
@@ -156,6 +167,8 @@ def raise_for_status(self, exit_status: int, stderr: bytes, context=None) -> Non
         if context and self.do_xcom_push:
             ti = context.get("task_instance")
             ti.xcom_push(key="ssh_exit", value=exit_status)
+        if self.skip_on_exit_code is not None and exit_status in self.skip_on_exit_code:
+            raise AirflowSkipException(f"SSH command returned exit code {self.skip_on_exit_code}. Skipping.")
```
Comment (Lee-W) [inline airflow/providers/ssh/operators/ssh.py]:
```suggestion
            raise AirflowSkipException(f"SSH command returned exit code {exit_status}. Skipping.")
```

I think we might want to log the raised exit code instead of the allowed exit code

File: docs/apache-airflow/howto/operator/bash.rst [hunk 2]
```diff
@@ -79,7 +79,7 @@ Skipping
 
 In general a non-zero exit code produces an AirflowException and thus a task failure.  In cases where it is desirable
 to instead have the task end in a ``skipped`` state, you can exit with code ``99`` (or with another exit code if you
-pass ``skip_exit_code``).
+pass ``skip_on_exit_code``).
```
Comment (Taragolis) [inline docs/apache-airflow/howto/operator/bash.rst]: Better to move it into the separate PR, because BashOperator is part of Airflow core and it has a different release cycle. It will make release managers happier 😉

File: airflow/providers/ssh/operators/ssh.py (hunk 1 above)
Comment (dolfinus) [reply airflow/providers/ssh/operators/ssh.py]: Fixed

File: docs/apache-airflow/howto/operator/bash.rst (hunk 2 above)
Comment (dolfinus) [reply docs/apache-airflow/howto/operator/bash.rst]: Ok

**Review by Taragolis — APPROVED** [review]

File: tests/providers/ssh/operators/test_ssh.py [hunk 3]
```diff
@@ -203,6 +203,37 @@ def test_ssh_client_managed_correctly(self):
...
+            ssh_hook=self.hook,
+            command=command,
+            **extra_kwargs,
+        )
+
+        if expected_exc is None:
+            operator.execute({})
+        else:
+            with pytest.raises(expected_exc):
+                operator.execute({})
```
Comment (josh-fell) [inline tests/providers/ssh/operators/test_ssh.py:237]:
```suggestion
        with pytest.raises(expected_exc):
            operator.execute({})
```
It's a nit really, but you don't need the branch here.

**Review by josh-fell — APPROVED** [review]

File: airflow/providers/ssh/operators/ssh.py [hunk 4]
```diff
@@ -156,6 +167,8 @@ def raise_for_status(self, exit_status: int, stderr: bytes, context=None) -> Non
         if context and self.do_xcom_push:
             ti = context.get("task_instance")
             ti.xcom_push(key="ssh_exit", value=exit_status)
+        if self.skip_on_exit_code is not None and exit_status in self.skip_on_exit_code:
```
Comment (hussein-awala) [inline airflow/providers/ssh/operators/ssh.py]: It will never be None:
```python
self.skip_on_exit_code = (
      skip_on_exit_code
      if isinstance(skip_on_exit_code, Container)
      else [skip_on_exit_code]
      if skip_on_exit_code
      else []
  )
```
```suggestion
        if exit_status in self.skip_on_exit_code:
```

**Review by hussein-awala — APPROVED** [review]
Nit

**Review by Lee-W — APPROVED** [review]

File: tests/providers/ssh/operators/test_ssh.py (hunk 3 above)
Comment (dolfinus) [reply tests/providers/ssh/operators/test_ssh.py:237]: There are test cases with exit code 0 which does not raise exception. But passing `None` or empty tuple to `pytest.raises` is not supported:
`ValueError: Expected an exception type or a tuple of exception types, but got None. Raising exceptions is already understood as failing the test, so you don't need any special code to say 'this should never raise an exception'.`
`ValueError: Expected an exception type or a tuple of exception types, but got (). Raising exceptions is already understood as failing the test, so you don't need any special code to say 'this should never raise an exception'.`

File: airflow/providers/ssh/operators/ssh.py (hunk 4 above)
Comment (dolfinus) [reply airflow/providers/ssh/operators/ssh.py]: Fixed

**Review by potiuk — APPROVED** [review]
